tests/5.1/masked/test_masked_filter.F90

```diff
@@ -0,0 +1,54 @@
+!/===--- test_masked_filter.F90 -------------------------------------===//
+!
+! OpenMP API Version 5.0 Nov 2018
+!
+! This test utilizes the masked construct to ensure that a section of code
+! only runs on a certain thread. When the filter clause is not present, the
+! primary thread runs the structured block.
+!
+!/===----------------------------------------------------------------------===//
+
+#include "ompvv.F90"
+
+#define N 1024
+
+PROGRAM test_masked
+  USE iso_fortran_env
+  USE ompvv_lib
+  USE omp_lib
+  implicit none
+
+  OMPVV_TEST_OFFLOADING
+
+  OMPVV_TEST_VERBOSE(masked() .ne. 0)
+
+  OMPVV_REPORT_AND_RETURN()
+
+CONTAINS
+  INTEGER FUNCTION masked()
+    INTEGER:: errors = 0
+    INTEGER:: total = 10
+    INTEGER:: ct = 0
+    INTEGER, DIMENSION(N):: x
+    INTEGER:: i
+
+    OMPVV_INFOMSG("test_masked")
+
+    DO i = 0, N
+       x(i) = 0
+    END DO
+    threads = omp_get_num_threads()
```

Comment:
See comments for the other Fortran file – all comments should also apply here.

However, with `num_threads(1)`, the testcase will deadlock as `filter(3)` will never become true. Thus, here, `num_threads(OMPVV_NUM_THREADS_HOST)` (directly or via assignment to `threads`) is required.